src/common/k8s-api/endpoints/ingress.api.ts

```diff
@@ -153,10 +154,11 @@ export class Ingress extends KubeObject {
     return routes;
   }
 
-  getServiceNamePort() {
-    const { spec } = this;
-    const serviceName = spec?.defaultBackend?.service.name ?? spec?.backend?.serviceName;
-    const servicePort = spec?.defaultBackend?.service.port.number ?? spec?.defaultBackend?.service.port.name ?? spec?.backend?.servicePort;
+  getServiceNamePort(): IExtensionsBackend {
+    const { spec: { backend, defaultBackend } = {}} = this;
+
+    const serviceName = defaultBackend?.service?.name ?? backend?.serviceName;
+    const servicePort = defaultBackend?.service?.port.number ?? defaultBackend?.service?.port.name ?? backend?.servicePort;
```

Comment:
where exactly was it crashing?

Reply:
It was crashing because `defaultBackend` was defined but `defaultBackend.service` was not, instead `defaultBackend.resource` was.